app/models/concerns/taggable.rb

```diff
@@ -0,0 +1,16 @@
+module Taggable
+  extend ActiveSupport::Concern
+
+  included do
+    has_many :taggings, as: :taggable, dependent: :destroy
+    has_many :tags, through: :taggings, source: :tag
+
+    scope :by_tags, ->(tags) { left_joins(:tags).where(tags: {name: tags}) }
```

Comment:
Can we name the parameter `tag_names`? It was a bit confusing till I saw where it was used.

Reply:
Fixed in 4561a63a145205bdfec559db2d394d6d4991ea76.